### ISSUE TITLE
Stylelint Issue

### ISSUE THREAD

**Issue by zslabs** [shellscape/postcss-less#81]
Hi,
Using the latest stylelint and postcss-less in a project, I'm seeing a similar error as noted in https://github.com/stylelint/stylelint/pull/2637#issuecomment-308471867

You may already be aware of this -- but is this more on their side or this one? If there's something I can help test I'm more than happy to help.

**Comment by shellscape** [comment]
without more discussion on that thread it's hard to tell. Given that master of stylelint was on `"postcss-less": "^0.14.0",` it's not a surprise that things could be breaking.

**Comment by jwilsson** [comment]
I think what's needed there is a [custom stringifier](https://github.com/shellscape/postcss-less#stringifying), as we did in `lesshint`. I'll let the `stylelint` team know about it too.

**Comment by zslabs** [comment]
I needed to add `import stringify from 'postcss-less/dist/less-stringify';` to be able to find `less-stringify` (as the path in the example didn't work), but Im' still getting the same error.

```
import gulp from 'gulp';
import gulpLoadPlugins from 'gulp-load-plugins';
import reporter from 'postcss-reporter';
import stylelint from 'stylelint';
import less from 'postcss-less';
import stringify from 'postcss-less/dist/less-stringify';

const $ = gulpLoadPlugins();

/**
 * Run styles through `lesshint`
 * @return {stream}
 */
export default function linter() {
  return gulp.src('./src/less/**/*.less')
  .pipe($.postcss(
    [
      stylelint(),
      reporter({ clearMessages: true }),
    ],
    {
      syntax: less,
      stringifier: stringify,
    },
  ));
}
```

**Comment by jwilsson** [comment]
@zslabs yep, I started poking at it in `stylelint` and there's more work needed than just that I'm afraid.

**Comment by shellscape** [comment]
@zslabs @jwilsson did I royally hose something in this module that the tests didn't cover, or is this just stylelint needing to make changes for the current version?

**Comment by jwilsson** [comment]
@shellscape No, I think we're good 😃 

There was a issue with stringifying `Import` nodes but I think I've found a solution for it which will remove the requirement for passing custom stringifiers too (I'll submit a PR for it shortly). 

The other issues are just small ones, changes for `rule.empty` etc. and some issue with semicolons in `node.raws` which I'm currently investigating.

**Comment by jwilsson** [comment]
This issue should be resolved with stylelint `8.0`.
